src/oasis-nexus/api.ts

```diff
@@ -243,8 +248,14 @@ export const useGetRuntimeTransactions: typeof generated.useGetRuntimeTransactio
           return {
             ...data,
             transactions: data.transactions.map(tx => {
+              const oasisAddress =
+                params?.rel && isValidEthAddress(params?.rel)
+                  ? getOasisAddressOrNull(params.rel)
+                  : params?.rel
+
               return {
                 ...tx,
+                to_eth: !tx.to_eth && tx.to === oasisAddress ? params?.rel : tx.to_eth,
```

Comment:
I would refactor this as:
```ts
const fallbackMaybeEthAddresses = [params?.rel]
const fallbackEthAddressesMap: { [oasis1Address: string]: `0x${string}` } =
  Object.fromEntries(
    fallbackMaybeEthAddresses.filter(isValidEthAddress).map((eth) => [getOasisAddressOrNull(eth), toChecksumAddress(eth)])
  )

return {
  ...tx,
  to_eth: tx.to_eth ?? fallbackEthAddressesMap[tx.to],
```

Reply:
if address map is preferable way than we don't need an array, fromEntries, filter, map as we deal with a single address.  Added a new commit with helper function.

Reply:
Hm, looks like we really don't have any EVM queries that support multiple address fields. nevermind